lib/src/layer/tile_layer.dart

```diff
@@ -946,6 +946,8 @@ class _TileLayerState extends State<TileLayer> with TickerProviderStateMixin {
           options.tileProvider.getImage(_wrapCoords(coords), options),
       tileReady: _tileReady,
     );
+
+    tile.loadTileImage();
```

Comment:
(nitpick) This code could be use the cascade operator to simplify:

```
_tiles[tileCoordsToKey] = Tile(
  // ...
)..loadTileImage();
```

Reply:
I tried it but didn't work. I'll give it another try maybe hotreload bugged.

Reply:
I retried it didn't work. I think this happens under hood when cascade operator is used.
1. Tile object created
2. `loadTileImage` method called on Tile
3. Tile added to `_tiles[tileCoordsToKey]`

With temp variable this happens:
1. Tile object created
2. Tile added to `_tiles[tileCoordsToKey]`
3. `loadTileImage` method called on Tile

So even if `loadTileImage`'s callback is synchronous we are sure it is already in `_tiles` map.